lib/uri.js

```diff
@@ -118,23 +128,33 @@ URI.prototype.expand = function(params) {
                 if (segmentValue === undefined) {
                     if (segment.modifier) {
                         // Okay to end the URI here
-                        return new URI(res);
+                        uri = new URI(res);
+                        uri._pathMetadata = pathMetadata;
+                        return uri;
                     } else {
                         throw new Error('URI.expand: parameter ' + segment.name + ' not defined!');
                     }
                 }
             }
-            if (Array.isArray(segmentValue)) {
-                res = res.concat(segmentValue);
+
+            if (segment.modifier === '+') {
+                // Res will become a path array, so we must split path elements
+                var splitPath = ('' + segmentValue).split('/');
+                res = res.concat(splitPath);
+                for (var j = res.length - 1; j < res.length - 1 + splitPath.length; j++) {
```

Comment:
My point is that `pathMetadata[j]` will always be set for values `j, j > path_length`, which seems rather fishy.